reactioncommerce/reaction-next-starterkit
# feat: Refresh access tokens on 401 errors

Resolves #350
Type: feature

## Issue
Get a new access token from Hydra.

## Solution
- Setup of a CORS-enabled endpoint in Identity provider that talks to Hydra to retrieve new tokens (reactioncommerce/reaction#4743)
- Update Apollo link setup to catch 401s
- Use endpoint created within IDP in apollo-client to get new tokens
- Write new tokens received to cookie 

## Breaking changes
None


## Testing
1. Start Starterkit on this branch, and Reaction API on this PR reactioncommerce/reaction#4743
2. Run Hydra on master. Set this ENV in the docker-compose yaml before starting the service: ACCESS_TOKEN_LIFESPAN=1m. This will reduce the wait time. The tokens will expire in 60s.
3. Load up a page. Leave it for a while for the token to expire. 
4. Return back to the page, and browser (e.g make a view a product, pause, then come back to add to cart. Try different approaches).
5. Note if there are any token denied errors or 401s. There should be none
